src/engine/qcommon/q_math.cpp

```diff
@@ -1510,7 +1510,7 @@ float VectorDistanceSquared( vec3_t v1, vec3_t v2 )
 // *INDENT-OFF*
 void MatrixIdentity( matrix_t m )
 {
-	memcpy( m, matrixIdentity, sizeof( matrix_t ) );
+	MatrixCopy( matrixIdentity, m );
```

Comment:
One thing I like with `memcpy` is that compilers usually know the best way to copy and use intrinsics as much as possible, usually even in debug mode, while you need some levels of compiler optimizations to get `MatrixCopy` vectorized by the compiler.

I wonder if we can have a custom `memcpy` with an assert it's not null (for non-user provided data), and one with a check otherwise, when `std::copy_n` is not available.

Reply:
That UB problem of `memcpy` is very annoying because `memcpy` is very efficient and is assumed to do the best for the task…

Reply:
Well, MatrixCopy still does memcpy :) This case is not one of the risky ones as the array is fixed-length. Here I'm really just trying to reduce the number of times `memcpy` appears in the source.

Also `std::copy` and friends are expected to shell out to memcpy when applicable (nonzero length, struct is POD or whatever)

Reply:
> Well, MatrixCopy still does memcpy :)

Ahah, yes, that's true for this one, I'm even the one who did it: d8089d6f8e62cbc7e3bb4581d21272d1f9a41130 😄️